src/util.test.ts

```diff
@@ -1,57 +1,105 @@
-    ])
-    assert.deepEqual(pipeline, [
-      {
-        $unset: [
-          'fullDocument.documents.agenda.parsedText',
-          'updateDescription.updatedFields.documents.agenda.parsedText',
-          'fullDocument.documents.agenda.contentType',
-          'updateDescription.updatedFields.documents.agenda.contentType',
-          'fullDocument.createdAt',
-          'updateDescription.updatedFields.createdAt',
-        ],
-      },
-      {
-        $set: {
-          'updateDescription.updatedFields': {
-            $arrayToObject: {
-              $filter: {
-                input: { $objectToArray: '$updateDescription.updatedFields' },
-                cond: {
-                  $regexMatch: {
-                    input: '$$this.k',
-                    regex:
-                      '^(?!documents\\.agenda\\.parsedText|documents\\.agenda\\.contentType)',
+    test('should generate pipeline from omit with dotted fields', () => {
+      const pipeline = generatePipelineFromOmit([
+        'documents.agenda.parsedText',
+        'documents.agenda.contentType',
+        'createdAt',
+      ])
+      assert.deepEqual(pipeline, [
+        {
+          $unset: [
+            'fullDocument.documents.agenda.parsedText',
+            'updateDescription.updatedFields.documents.agenda.parsedText',
+            'fullDocument.documents.agenda.contentType',
+            'updateDescription.updatedFields.documents.agenda.contentType',
+            'fullDocument.createdAt',
+            'updateDescription.updatedFields.createdAt',
+          ],
+        },
+        {
+          $set: {
+            'updateDescription.updatedFields': {
+              $arrayToObject: {
+                $filter: {
+                  input: { $objectToArray: '$updateDescription.updatedFields' },
+                  cond: {
+                    $regexMatch: {
+                      input: '$$this.k',
+                      regex:
+                        '^(?!documents\\.agenda\\.parsedText|documents\\.agenda\\.contentType)',
+                    },
                   },
                 },
               },
             },
           },
         },
-      },
-    ])
+      ])
+    })
+  })
+  describe('removeUnusedFields', () => {
+    test('should remove all unneeded fields', () => {
+      const schema = {
+        bsonType: 'object',
+        required: ['a', 'b'],
+        additionalProperties: false,
+        properties: {
+          a: {
+            title: 'An array of strings',
+            bsonType: 'array',
+            items: {
+              bsonType: 'string',
+              title: 'A string',
+            },
+          },
+          b: {
+            description: 'foo or bar',
+            bsonType: 'string',
+            enum: ['foo', 'bar'],
+            oneOf: [
+              { bsonType: 'string', const: 'foo' },
+              { bsonType: 'string', const: 'bar' },
+            ],
+          },
+        },
+      }
+      removeUnusedFields(schema)
```

Comment:
It would be nice if this function didn't have side effects. It could work like this:

```typescript
assert.deepEqual(
  removeUnusedFields(schema),
  { /* what you have below */ }
)
```

Reply:
It's intentional that `removeUnusedFields` mutates. There's no reason to deep-clone the object. The function is internal only. Deep-cloning would only decrease performance.

Reply:
Makes sense.